tql_macros/src/analyzer/aggregate.rs

```diff
@@ -166,7 +166,8 @@ fn binary_expression_to_aggregate_filter_expression(expr1: &Expression, op: &Bin
 
 /// Check that an aggregate field exists.
 fn check_aggregate_field<'a>(identifier: &str, aggregates: &'a [Aggregate], position: Span, errors: &mut Vec<Error>) -> Option<&'a Aggregate> {
-    let result = aggregates.iter().find(|aggr| aggr.result_name.as_ref().map(|name| name.as_ref()) == Some(identifier));
+    let matching = Some(identifier.to_string());
+    let result = aggregates.iter().find(|aggr| aggr.result_name.as_ref().map(|name| name.to_string()) == matching);
```

Comment:
Was it causing a compile-time error to use `as_ref()`, because it seems to me we should be able to compare to `&str` without converting them to `String`?

Reply:
for `let matching = Some(identifier.to_string());` it might be true, i will check it out later. However, `name.to_string()` is used because name is `syn::Ident`, which is the same case as bellow

Reply:
`Ident` has an implementation that might be useful: https://docs.rs/syn/0.15.29/syn/struct.Ident.html#impl-PartialEq%3CT%3E

Reply:
Ok, i updated this one to use PartialEq. As a note: it probably still uses `to_string()` internally: `proc_macro::Ident` doesn't implement this trait, so `proc_macro2` tries to emulate this using `to_string()` : https://github.com/alexcrichton/proc-macro2/blob/35d24c390b6e3cd1cff5f2b4e1b50a91004ee76c/src/wrapper.rs#L746 .